docs/DeveloperGuide.md

```diff
@@ -8,23 +8,119 @@
+
+### [Proposed] EditIncomeCommand/EditExpenseCommand
+The proposed edit income command is facilitated by `Parser`, `EditIncomeCommand`, `IncomeList`, while the proposed edit expense command is facilitated by `Parser`, `EditExpenseCommand` and `ExpenseList`. 
+
+Note that below highlights the implementation of the edit income command, with edit expense command following the same implementation.
+
+Given below is how the edit income mechanism works at each step:
+1. The user types in the command for edit income with the necessary arguments given to indicate which income record to edit and what fields to edit for the chosen income record.
+2. The line inputted by the user will then be parsed by `Parser`, where it will check if the input contains a valid index with valid fields to edit. If they are valid, it will return `EditIncomeCommand`. Else it will throw a `ChChingException` indicating an error in the input from the user.
+3. `EditIncomeCommand` will then perform its `execute` method, where for each field to be edited, it will call `editIncome` method from the `ExpenseList` class. This would then update the required income record accordingly.
+
+The following sequence diagram shows how the edit income command works:
+<br> ![edit income sequence diagram](images/EditIncomeCommand_sequence_diagram.png)
+
+The edit expense command works in a similar way, with its sequence diagram as shown:
+<br> ![edit expense sequence diagram](images/EditExpenseCommand_sequence_diagram.png)
+
+The following activity diagram summarises what happens when a user executes edit income command:
+<br> ![edit income activity diagram](images/EditIncomeCommand_activity_diagram.png)
+<br> Note that edit expense command produces the same activity diagram.
+
+**Design Considerations**
+<br>The following are the design alternatives we considered for edit income/expense command:
+
+* **Alternative 1 (current choice):** Edit only the specific fields indicated in the input arguments.
+    * Pros: Easier for the user to input an edit.
+    * Cons: May introduce more bugs.
+* **Alternative 2:** Edit to require user to rewrite all its fields.
+    * Pros: Easier to implement.
+    * Cons: Not any easier than having the user to just delete and add new expense/income.
+
+### ListExpenseCommand
+The listExpenseCommand is facilitated by ```Parser```, ```ListExpenseCommand``` and ```ExpenseList```.
+
+1. The user inputs the command top list expense. This input is handled by```Parser``` which returns the
+   ```listExpenseCommand``` if successful.
+2. ```ListExpenseCommand``` will then call its ```execute``` method
+   which makes use of ```System``` to print a new line ```"Expenses:"```.
+3. The ```expenseList``` method ```printExpenseList```
+   is called, which iterates through the expenseList, ```expenses``` and prints the index as well as a completed string of
+   expenses in ```expenses```.
+
+![ListExpenseCommand](images/ListExpenseCommand_Sequence_Diagram.png)
```

Comment:
For this diagram:
1. This image is named as a sequence diagram when it is actually a class diagram
2. Other than showing ListExpenseCommand class inherits from Command class, this does not add much significance at this point of the dg
3. Possible to move this class diagram up above

Reply:
The execute portion of the class diagram is also incorrect. It should be something like execute( ... ): void.